packages/desktop-client/src/components/mobile/budget/BudgetTable.jsx

```diff
@@ -1564,6 +1572,28 @@ export function BudgetTable({
     'mobile.showSpentColumn',
   );
 
+  const scrollToPosition = () => {
+    sessionStorage.setItem(
+      'scrollPosition',
+      document.getElementById('scrollableDiv').scrollTop,
+    );
+    sessionStorage.setItem('scrollPositionSetBySpentColumn', true);
+  };
+
+  useEffect(() => {
+    const savedPosition = parseInt(
+      sessionStorage.getItem('scrollPosition'),
+      10,
+    );
+    const scrollPositionSetBySpentColumn = sessionStorage.getItem(
+      'scrollPositionSetBySpentColumn',
+    );
+    if (savedPosition && scrollPositionSetBySpentColumn) {
+      document.getElementById('scrollableDiv').scrollTop = savedPosition;
+      sessionStorage.removeItem('scrollPositionSetBySpentColumn');
+    }
+  }, []);
+
```

Comment:
I wonder if it is possible to pull the logic for return scrolling into a react component, and then you can just wrap things with that component? Would avoid the duplicated/repeated functionality, and also allow for easily adding it to other places (and I have some in mind).

Reply:
> I wonder if it is possible to pull the logic for return scrolling into a react component, and then you can just wrap things with that component? Would avoid the duplicated/repeated functionality, and also allow for easily adding it to other places (and I have some in mind).

I gave it a try, and I cannot figure out how to accomplish this. If you want the feature above, review this feature as is, or we can close it.

Thank you.

Reply:
Let's just merge this as is and then someone can refactor the common code into a shared component.  And then once we have that, we can add it other places more easily.